src/cpr_data_access/utils.py

```diff
@@ -86,3 +86,29 @@ def dig(obj: Union[list, dict], *fields: Any, default: Any = None) -> Any:
         elif not obj:
             return default
     return obj
+
+
+def unflatten_json(data: dict) -> dict:
+    """
+    Unflatten a dictionary with keys that are dot-separated strings.
+
+    I.e. metadata.data respresents {"metadata": {"data": {}}}
+    """
+    unflattened = {}
+    for key, value in data.items():
+        parts = key.split(".")
+        current = unflattened
+        for part in parts[:-1]:
+            current = current.setdefault(part, {})
+        current[parts[-1]] = value
+    return unflattened
+
+
+def remove_if_all_nested_vals_none(data: dict, key: str) -> dict:
+    """Remove an item if it's a dict with all values in the nested dictionary are None."""
+    if key not in data:
+        return data
+    if isinstance(data[key], dict):
+        if all(value is None for value in data[key].values()):
+            data.pop(key)
```

Comment:
Any reason for keeping the field if the value is `None`? More intuitive behaviour for me would be removing. I see that you have a test covering that case, so wondering if that's for any particular reason.

Reply:
Not sure I follow, in this function we are removing the key if all the values of the the dict are None. 

E.g. {"key": {"a": None, "b": None}} -> {}